### ISSUE TITLE
Support adding apps to multiple Proxy / VPN profiles

### ISSUE THREAD

**Issue by ignoramous** [celzero/rethink-app#1023]
> The ability to have multiple wireguard VPN  tunnels running at the same time is actually quite convenient so my suggestion is if multiple profile are active at the same time and lets say the same app was active on different profiles the traffic should be routed to the top wireguard profile ( you could give the option to order the wireguard profiles so rethink would route the app traffic on the first wireguard profile the app is listed in and ignore the other profiles.

_Originally posted by @mecoolk1 in https://github.com/celzero/rethink-app/issues/1019#issuecomment-1692607549_

**Comment by kamzil** [comment]
IMO instead of following some specific order and only support one tunnel at a time per app, it should respect the Allowed IPs settings for each WireGuard profile such that it allows split-tunneling the traffic from the same app to different tunnels based on the Allowed IPs configs. Isn't this what the "Split" tag next to the profile name means anyway? This would allow using one WG for stuff in your LAN and another one for stuff in public internet. Perhaps even a third one for e.g. work. (Note that I'm using older version of this app until it gets more stable. Great job guys! Am following with excitement what's happening here.)

**Comment by ignoramous** [comment]
> IMO instead of following some specific order and only support one tunnel at a time per app, it should respect the Allowed IPs settings for each WireGuard profile such that it allows split-tunneling the traffic from the same app to different tunnels based on the Allowed IPs configs. Isn't this what the "Split" tag next to the profile name means anyway? 

Yep. Already implemented this and will be in the next version `v055v` (for WireGuard but not other proxies).

**Comment by ignoramous** [comment]
Shipped in `v055v` but probably `v055y` is a much more bug free (see issues linked below) implementation of this feature.

- #2854
- #2831
